### ISSUE TITLE
ERROR: Could not open a Session with StatusCode BadServiceUnsupported

### ISSUE THREAD

**Issue by jadhavshital** [open62541/open62541#3447]
Hi,
I have built open62541 encrypted library on windows 10. I tried with examples given [here](https://github.com/open62541/open62541/tree/master/examples/encryption). For this I had created certificates using https://github.com/open62541/open62541/tree/master/tools/certs/create_self-signed.py script. And those examples worked fine.

But when I try to connect with Prosys OpcUa Simulation Server, I am getting below error - 
![image](https://user-images.githubusercontent.com/17235485/74723172-263b9b00-5260-11ea-90bf-7ff6f0e3721b.png)

My sample code is -
``	/* Load certificate and private key */
    UA_ByteString certificate = loadFile(cert_file_path.c_str());
    UA_ByteString privateKey = loadFile(key_file_path.c_str());

    /* Load the trustList. Load revocationList is not supported now */
    size_t trustListSize = 0;
    if (argc > MIN_ARGS)
        trustListSize = (size_t)argc - MIN_ARGS;
    UA_STACKARRAY(UA_ByteString, trustList, trustListSize);
    for (size_t trustListCount = 0; trustListCount < trustListSize; trustListCount++){
        trustList[trustListCount] = loadFile(argv[trustListCount + 4]);
        std::cout << " Trustlist [trustListCount] " << (trustList[trustListCount]).data << ", i = " << trustListCount << std::endl;
    }

    UA_ByteString* revocationList = NULL;
    size_t revocationListSize = 0;

    UA_Client* client = UA_Client_new();
    UA_ClientConfig* cc = UA_Client_getConfig(client);
    cc->securityMode = UA_MESSAGESECURITYMODE_SIGNANDENCRYPT;
    cc->securityPolicyUri = UA_STRING_ALLOC("http://opcfoundation.org/UA/SecurityPolicy#Basic256Sha256");
    UA_ClientConfig_setDefaultEncryption(cc, certificate, privateKey,
        trustList, trustListSize,
        revocationList, revocationListSize);

    UA_ByteString_clear(&certificate);
    UA_ByteString_clear(&privateKey);
    for (size_t deleteCount = 0; deleteCount < trustListSize; deleteCount++) {
        UA_ByteString_clear(&trustList[deleteCount]);
    }
   
    UA_StatusCode retval = UA_Client_connect(client, endpointUrl.c_str());``

Prosys simulation server - 
![image](https://user-images.githubusercontent.com/17235485/74723716-0bb5f180-5261-11ea-91a9-8b51437de075.png)
![image](https://user-images.githubusercontent.com/17235485/74723757-1a040d80-5261-11ea-920e-297149a703d1.png)
Certificate has been added to server -
![image](https://user-images.githubusercontent.com/17235485/74723802-2e480a80-5261-11ea-905b-a7f0e4fb3c82.png)

Kindly help!

**Comment by luiztsmendes** [comment]
I have a similar problem on Windows 10 platform when connecting to the Prosys OPC UA Simulation server (status code BadServiceUnsupported). However, the client example code I used was the simpler one provided in the open62514 tutorial "Building a Simple Client", which just connects to a server and reads the server's current time:

![open62541_1 0 3](https://user-images.githubusercontent.com/32816561/75061564-bc213100-54bf-11ea-8d4c-f587a3936331.jpg)

However, when using the same client example from the 0.3 branch it worked fine:

![open62541_0 3](https://user-images.githubusercontent.com/32816561/75061793-3ce02d00-54c0-11ea-98fc-c6bf3a083451.jpg)

Could it be a problem with the default client configuration? I noticed that the client configuration for the 0,3 branch is much simpler than the one from the master branch.

Anyway, thanks for this outstanding job of developing an open OPC UA library!
